src/main/lens-proxy.ts

```diff
@@ -127,8 +127,10 @@ export class LensProxy {
         this.retryCounters.delete(retryCounterId);
       }
 
-      if (!res.headersSent) {
-        res.flushHeaders();
+      if (!res.headersSent && req.url) {
+        const url = new URL(req.url, "http://localhost");
+
+        if (url.searchParams.has("watch")) res.flushHeaders();
       }
```

Comment:
```suggestion
      if (!res.headersSent && (new URL(req.url, "http://localhost")).searchParams.has("watch")) {
        res.flushHeaders();
      }
```

Reply:
I prefer my version of this, it's longer but easier to read 🙂